src/CatalogApplication/Models/README.md

```diff
@@ -0,0 +1,54 @@
+# UML Model For Tags/Ratings
+![alt text](https://i.imgur.com/eMo0eor.png "UML Diagram")
+
+# Tag Attributes
+```json
+Tag(
+    id:          int      # Generated by us somehow
+    name:        str      # The actual tag name
+    color:       int      # Hex value
+    note_id:     int      # Note ID associated with this tag
+    hidden:      bool     # Should the tag be shown
+    is_system:   bool     # Was this tag automatically added by us?
+    category:    str      # Engineering, Chemistry, CS, etc?
+)
+```
+
+## unique_id
+This is an integer generated by +1 increment
+
+## name
+A string representing the literal name of the tag i.e data-structures
+
+## color
+Hexadecimal integer value representing "good" or "bad" tag
+
+## note_id
+Integer note ID associated with the tag
+
+## hidden
+Boolean value determining whether or not tag should be shown
```

Comment:
Could you explain when you would have different values for `hidden` and `is_system`?

Reply:
Our idea behind this was that there could be a tag that gets put as hidden despite it being put on automatically by the system. For now these two values will be correlated, but for future implementations it might be nice to separate "tags put on by the system" and "hidden tags", because technically they don't have to be the same.